python/rikai/spark/functions/vision.py

```diff
@@ -104,7 +106,8 @@ def numpy_to_image(array: ndarray, uri: str) -> Image:
 
 @udf(returnType=ArrayType(ImageType()))
 def video_to_images(
-    video,
+    video: Union[VideoStream, YouTubeVideo],
+    output_uri: str,
```

Comment:
Discussed in another thread,  my main concern is that this parameter drags users from describing algorithms, to must be aware of infra details ("i.e., where the storage is"). This is a distraction to user's workflow. 

I'd encourage us to re-think how Image / Video asset class is implemented.

Reply:
Yeah, I totally agree. Previously we had talked about supporting and Image option that is just an in-memory ndarray and create an explicit externalization method that switches between what we have now for Image and that in-memory representation.
Once we have that, the user can choose to externalize the images, otherwise we store them directly with the image data embedded in the column.

The counter-argument to hiding the storage is that if you automatically stash it somewhere, it'll be very easy for the user to lose the actual data even though they may think they know exactly where they put the dataframe. 

I made #144 to keep track of it. My suggestion here would be to get this change in so this is at least a predictable/consistent feature. Then once we create the alternative Image class we can refactor this one to remove the uri (or make it optional).

Reply:
Make sense. Lets make things moving :)